llvm/include/llvm/CodeGen/MachineBasicBlock.h

```diff
@@ -1263,14 +1263,16 @@ class MachineBasicBlock
   /// MachineBranchProbabilityInfo class.
   BranchProbability getSuccProbability(const_succ_iterator Succ) const;
 
+  // Helper function for MIRPrinter.
+  bool canPredictBranchProbabilities() const;
```

Comment:
Why not just leave it there?

Reply:
Currently MIRPrinter is in `llvm` namespace so that it can be declared as a friend of `MachineBasicBlock` to allow it to access MachineBasicBlock::Probs (private) to compute canPredictBranchProbabilities. This remove that coupling by moving just this function to MachineBasicBlock, and that allows MIRPrinter to be anonymous and not a friend.